### PR TITLE
checks audience in everyone else rule

### PR DESCRIPTION
**Pull request by rashidsp**

### CODE REVIEW

**Comment by rashidsp** [comment]
@oakbani Are we using any linter for PHP?
And i have not changed anything in files with Coverage Reduction.

File: src/Optimizely/DecisionService/DecisionService.php [hunk 1]
```diff
@@ -304,35 +304,29 @@ public function getVariationForFeatureRollout(FeatureFlag $featureFlag, $userId,
...
+            }
+        } else{
             $this->_logger->log(
                 Logger::DEBUG,
-                "User '{$userId}' was excluded from the 'Everyone Else' rule for feature flag"
+                sprintf("User '%s' did not meet the audience conditions to be in rollout rule '%s'.", $userId, $experiment->getKey())
             );
-            return null;
         }
+        return null;
```
Comment (oakbani) [inline src/Optimizely/DecisionService/DecisionService.php]: Above code is difficult to read. 
Avoid else block. 
If audience match fails, log and return immediately.

File: tests/DecisionServiceTests/DecisionServiceTest.php [hunk 2]
```diff
@@ -1125,4 +1084,62 @@ public function testGetVariationForFeatureRolloutWhenUserDoesNotQualifyForAnyTar
...
+        
+        $this->loggerMock->expects($this->at(2))
+            ->method('log')
+            ->with(
+                Logger::DEBUG,
+                "User 'user_1' did not meet the audience conditions to be in rollout rule '{$experiment2->getKey()}'."
+        );    
+
+        $this->assertEquals(
+            null,
```
Comment (oakbani) [inline tests/DecisionServiceTests/DecisionServiceTest.php]: nit - assertNull

File: tests/DecisionServiceTests/DecisionServiceTest.php [hunk 3]
```diff
@@ -1125,4 +1084,62 @@ public function testGetVariationForFeatureRolloutWhenUserDoesNotQualifyForAnyTar
...
+        
+        // Set an AudienceId for everyone else/last rule so that user does not qualify for audience
+        $experiment2->setAudienceIds(["11154"]);
+        $expected_variation = $experiment2->getVariations()[0];
+        $expected_decision = new FeatureDecision(
+            $experiment2,
+            $expected_variation,
+            FeatureDecision::DECISION_SOURCE_ROLLOUT
+        );
+
```
Comment (oakbani) [inline tests/DecisionServiceTests/DecisionServiceTest.php]: Unused code:

```
$expected_variation = $experiment2->getVariations()[0];
+        $expected_decision = new FeatureDecision(
+            $experiment2,
+            $expected_variation,
+            FeatureDecision::DECISION_SOURCE_ROLLOUT
+        );
```

File: tests/DecisionServiceTests/DecisionServiceTest.php [hunk 4]
```diff
@@ -1125,4 +1084,62 @@ public function testGetVariationForFeatureRolloutWhenUserDoesNotQualifyForAnyTar
...
+        // Provide null attributes so that user does not qualify for audience
+        $user_attributes = [];
+        $this->decisionService = new DecisionService($this->loggerMock, $this->config);
+        $bucketer = new \ReflectionProperty(DecisionService::class, '_bucketer');
+        $bucketer->setAccessible(true);
+        $bucketer->setValue($this->decisionService, $this->bucketerMock);
+
+        // // Expect bucket never called for the everyone else/last rule.
+        $this->bucketerMock->expects($this->exactly(0))
```
Comment (oakbani) [inline tests/DecisionServiceTests/DecisionServiceTest.php]: nit - $this->never()

File: tests/DecisionServiceTests/DecisionServiceTest.php [hunk 5]
```diff
@@ -1125,4 +1084,62 @@ public function testGetVariationForFeatureRolloutWhenUserDoesNotQualifyForAnyTar
...
+            ->method('bucket')
+            ->willReturn($expected_variation);
```
Comment (oakbani) [inline tests/DecisionServiceTests/DecisionServiceTest.php]: Why will it return if it's never called?

**Pull request closed without merging**